### PR TITLE
Upgrade matrix to 5.1.0

### PR DESCRIPTION
**Pull request by andrevmatos**
And get rid of custom type definitions for `matrix-js-sdk`, favoring the `@types/` one.
Some adaptation and better error handling were needed too, for matrix rooms operations and maintainance.
Includes rewrite of `package-lock.ts`s.

### CODE REVIEW

File: raiden-ts/tests/e2e/provider.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export class TestProvider extends Web3Provider {
           seed: 'testrpc_provider',
           network_id: 1338,
           db: memdown(),
-          logger: console,
+          // logger: console,
```
Comment (kelsos) [inline raiden-ts/tests/e2e/provider.ts]: Why comment instead of removing?

Comment (andrevmatos) [reply raiden-ts/tests/e2e/provider.ts]: I uncomment eventually to debug e2e tests, and some time seems like I forgot to comment out again 😅

Comment (kelsos) [reply raiden-ts/tests/e2e/provider.ts]: Can we make it activatable by some environment variable or something? :P

Comment (andrevmatos) [reply raiden-ts/tests/e2e/provider.ts]: Haha, you guys hate logs, this is more for local testing & test debugging in some very specific cases, but if someone feels the need, we can improve it at some point ;)

Comment (kelsos) [reply raiden-ts/tests/e2e/provider.ts]: Well not hate. I like them when there is an actual issue. But they come with a performance overhead and they can be overwhelming when they are too verbose. 

In the context ofthis comment though, I would say not as much as I hate commented code :P 

The environment variable here was an alternative to commenting and uncommenting the logger code for the tests.

**Pull request merged**